### PR TITLE
Made Lua/solution_1 faithful

### PR DESCRIPTION
**Pull request by greatwolf**
## Description
This PR fixes up Lua solution_1 to satisfy the faithful requirement by making it more OOP's. The changes are mostly straightforward. 

#### Quick rundown of changes:

 - The sieve state is now encapsulated into a lua table to represent it as an 'object'.
 - Created `PrimeSieve` table and placed the free functions into it so it acts like a 'class' definition.
 - The various free functions were also updated to accept an implicit `self` as first parameter effectively turning them into methods that operate on a given object.
 - Lua's metatable mechanism is used to tie the methods to a given object(table) so it can be operated on with object-like syntax. 
 - Renamed `setup` function to `new` so it's more appropriate. New instantiations can be created with `PrimeSieve:new(size)`.
 - Updated `printResults` function to just print the output string directly so it more closely resembles Dave's original implementation.

## Contributing requirements

* [x] I read the contribution guidelines in CONTRIBUTING.md.
* [x] I placed my solution in the correct solution folder.
* [x] I added a README.md with the right badge(s).
* [x] I added a Dockerfile that builds and runs my solution.
* [x] I selected `drag-race` as the target branch.
* [x] All code herein is licensed compatible with BSD-3.

### CODE REVIEW

**Review by rbergen — COMMENTED** [review]
I believe the "Faithful" badge in the README.md also needs to be updated (to "yes-green")

**Comment by greatwolf** [comment]
whoops forgot that one. ok done.

File: PrimeLua/solution_1/README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Lua solution based on sources from seanbouk
 
 ![Algorithm](https://img.shields.io/badge/Algorithm-base-green)
-![Faithfulness](https://img.shields.io/badge/Faithful-no-yellowgreen)
+![Faithfulness](https://img.shields.io/badge/Faithful-yes-yellowgreen)
```
Comment (rbergen) [inline PrimeLua/solution_1/README.md]: We're nearly there. :) Please change -yellowgreen to -green.

Comment (greatwolf) [reply PrimeLua/solution_1/README.md]: that should do it.

**Pull request merged**